base/docs/basedocs.jl

```diff
@@ -1868,6 +1868,29 @@ typeassert
 
 The syntax `a.b` calls `getproperty(a, :b)`.
 
+For example, to run custom code to evaluate
+a property by a specific symbol name, you can
+override this method and create a specific
+if-branch as shown below.
+
+```
+struct MyType
+    x
+end
+
+function Base.getproperty(obj::MyType, sym::Symbol)
+    if sym === :special
+        obj.x + 1
+    else
+        # fallback to getfield
+        getfield(obj, sym)
+    end
+end
+
+obj = MyType(1)
+obj.special
+```
+
```

Comment:
Maybe skip the introductory text and just add `# Examples` as we do for other docstrings? Also, maybe write this as a `jldoctest`. Maybe something like
````
# Examples
```jldoctest
julia> struct MyType
           x
       end

julia> function Base.getproperty(obj::MyType, sym::Symbol)
           if sym === :special
               return obj.x + 1
           else # fallback to getfield
               return getfield(obj, sym)
           end
       end

julia> obj = MyType(1);

julia> obj.special
2

julia> obj.x
1
```
````